docs/STRATEGY.md

```diff
@@ -0,0 +1,168 @@
+# Migration Strategy
+
+## Requirements
+
+However, in order for a migration script to use this framework, it needs to instantiate
+a migration strategy, fulfilling the following requirements:
+
+1. Needs to respond to `#process(item)`. It should implement a `#process(item)` method that
```

Comment:
After reading this I feeling that the  explanation of criteria should come query. To my understanding you create a criteria that gives you a collection then query is what action you want to take place on an element of the collection. having them the other way round is a bit confusing.